build/ccache/download_ccache.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import argparse
+import glob
+import os
+import sys
+import shutil
+
+import deps
+
+UNPACK_DIR = os.path.join(os.path.dirname(__file__), '_unpack')
+CCACHE_DIR = os.path.join(os.path.dirname(__file__), 'bin')
+
+
+def main():
+    args = parse_args()
+    if args.version == get_current_version():
+        return
+
+    if os.path.exists(CCACHE_DIR):
+        shutil.rmtree(CCACHE_DIR)
+
+    url = ('https://github.com/ccache/ccache/releases/download/'
+           f'v{args.version}/ccache-{args.version}-')
+    if sys.platform == 'linux':
+        url += 'linux-x86_64.tar.xz'
+    elif sys.platform == 'darwin':
+        url += 'darwin.tar.gz'
+    else:
+        url += 'windows-x86_64.zip'
```

Comment:
@mihaiplesa can you pls copy all three files to https://brave-build-deps-public.s3.brave.com/ccache/... ? 
https://github.com/ccache/ccache/releases/download/v4.8.3/ccache-4.8.3-darwin.tar.gz
https://github.com/ccache/ccache/releases/download/v4.8.3/ccache-4.8.3-linux-x86_64.tar.xz
https://github.com/ccache/ccache/releases/download/v4.8.3/ccache-4.8.3-windows-x86_64.zip

I'll replace URL here to point to our server.

Reply:
https://brave-build-deps-public.s3.brave.com/ccache/ccache-4.8.3-darwin.tar.gz
https://brave-build-deps-public.s3.brave.com/ccache/ccache-4.8.3-linux-x86_64.tar.xz
https://brave-build-deps-public.s3.brave.com/ccache/ccache-4.8.3-windows-x86_64.zip